packages/xstate-solid/src/useMachine.ts

```diff
@@ -0,0 +1,37 @@
+import type { AnyStateMachine, StateFrom } from 'xstate';
+import type { CheckSnapshot, RestParams } from './types';
+import { createService } from './createService';
+import { onCleanup, onMount } from 'solid-js';
+import type { InterpreterFrom, Prop } from 'xstate';
+import { deriveServiceState } from './deriveServiceState';
+import { createImmutable } from './createImmutable';
+
+type UseMachineReturn<
+  TMachine extends AnyStateMachine,
+  TInterpreter = InterpreterFrom<TMachine>
+> = [
+  CheckSnapshot<StateFrom<TMachine>>,
+  Prop<TInterpreter, 'send'>,
+  TInterpreter
+];
+
+export function useMachine<TMachine extends AnyStateMachine>(
+  machine: TMachine,
+  ...[options = {}]: RestParams<TMachine>
+): UseMachineReturn<TMachine> {
+  const service = createService(machine, options);
+
+  const [state, setState] = createImmutable(
+    deriveServiceState(service, service.getSnapshot())
```

Comment:
this should be fine (and will be in the future) but for the time being it's not safe as you could spawn some initial actors through this call, before the machine even starts.

I'd copy here what we do in `@xstate/react`:
https://github.com/statelyai/xstate/blob/3ee04b92d6fc44b1375bd2076557cf96ada98387/packages/xstate-react/src/useMachine.ts#L73-L79